ingest/ingest.py

```diff
@@ -0,0 +1,451 @@
+            self.batch = setup_batch(
+                title='Transfer 1KG data',
+                keep_scratch=False,
+                tmp_bucket=f'cpg-{sample_metadata_project}-{namespace}-tmp',
+                analysis_project=sample_metadata_project,
+            )
+        else:
+            self.batch = None
+        _call(f'mkdir -p {self.tmp_dir}')
+        _call(f'mkdir -p {self.resources_dir}')
+
+    def get_sample_id(self, row: Dict[str, any]) -> str:
+        return row['s']
+
+    def get_sample_meta(self, sample_id: str, row: GroupedRow) -> Dict[str, any]:
+        sample_meta = dict()
+        sample_meta['continental_pop'] = row['continental_pop']
+        sample_meta['fam_id'] = row['fam_id']
+        sample_meta['pat_id'] = row['pat_id']
+        sample_meta['mat_id'] = row['mat_id']
+        sample_meta['sex'] = row['sex']
+        return sample_meta
+
+    def get_sequence_meta(self, sample_id: str, row: GroupedRow) -> Dict[str, any]:
+        sequence_meta = dict()
+        if 'gvcf' in row and row['gvcf'] and row['gvcf'] != '-':
+            gvcf, variants_type = self.parse_file([row['gvcf']])
+            sequence_meta['gvcf'] = gvcf
+            sequence_meta['gvcf_type'] = variants_type
+        if 'cram' in row and row['cram'] and row['cram'] != '-':
+            reads, reads_type = self.parse_file([row['cram']])
+            sequence_meta['reads'] = reads
+            sequence_meta['reads_type'] = reads_type
+        return sequence_meta
+
+    def get_sequence_status(self, sample_id: str, row: GroupedRow) -> str:
+        return 'uploaded'
+
+    def find_files(self) -> str:
+        """
+        Finds GVCFs and CRAMs on Broad buckets, matches them with pedigree information,
+        makes a metadata TSV file andn returns a path to it
+        """
+        ped_path = self._download_ped()
+        src_gvcf_ls_path = self._save_bucket_ls(
+            ext=['.raw.vcf.gz', '.raw.g.vcf.gz'],
+            source_bucket=f'{SOURCE_BUCKET}/**',
+            label='source-gvcfs',
+        )
+        src_cram_ls_path = self._save_bucket_ls(
+            ext='.cram',
+            source_bucket=f'{SOURCE_BUCKET}/**',
+            label='source-crams',
+        )
+        output_path = join(self.tmp_dir, 'samples-raw.tsv')
+        if can_reuse(output_path, self.overwrite):
+            return output_path
+
+        ped_df = pd.read_csv(ped_path, sep='\t')
+        print(f'Found {len(ped_df)} sample in PED')
+        ped_row_by_sid = dict()
+        for _, ped_row in ped_df.iterrows():
+            s = ped_row['Individual.ID']
+            ped_row_by_sid[s] = ped_row
+
+        gvcf_by_sid = dict()
+        cram_by_sid = dict()
+        with open(src_gvcf_ls_path) as f:
+            for line in f:
+                if line.strip():
+                    source_gvcf_path = line.strip()
+                    sid = source_gvcf_path.split('/')[-1].split('.')[0]
+                    if sid in ['NA18874A', 'NA12546B', 'NA12830A']:
```

Comment:
Is there something specific about these sample Ids, do you mind adding a comment / article to link to?